e2e-tests/playwright/utils/ui-helper.ts

```diff
@@ -218,6 +218,61 @@ export class UIhelper {
     await expect(elementLocator).toBeVisible();
   }
 
+  async verifyTextInSelector(selector: string, expectedText: string) {
+    const elementLocator = this.page
+      .locator(selector)
+      .getByText(expectedText, { exact: true });
+
+    try {
+      await elementLocator.waitFor({ state: "visible", timeout: 10000 });
+      const actualText = (await elementLocator.textContent()) || "No content";
+
+      if (actualText.trim() !== expectedText.trim()) {
+        console.error(
+          `Verification failed for text: Expected "${expectedText}", but got "${actualText}"`,
+        );
+        throw new Error(
+          `Expected text "${expectedText}" not found. Actual content: "${actualText}".`,
+        );
+      }
+      console.log(
+        `Text "${expectedText}" verified successfully in selector: ${selector}`,
+      );
+    } catch (error) {
+      const allTextContent = await this.page
+        .locator(selector)
+        .allTextContents();
+      console.error(
+        `Verification failed for text: Expected "${expectedText}". Selector content: ${allTextContent.join(", ")}`,
+      );
+      throw error;
+    }
+  }
+
+  async verifyPartialTextInSelector(selector: string, partialText: string) {
+    try {
+      const elements = await this.page.locator(selector);
+      const count = await elements.count();
+
+      for (let i = 0; i < count; i++) {
+        const textContent = await elements.nth(i).textContent();
+        if (textContent && textContent.includes(partialText)) {
+          console.log(
+            `Found partial text: ${partialText} in element: ${textContent}`,
```

Comment:
@gustavolira could we use playwright assertions here instead of a try-catch block? using proper assertions would improve debugging and ensure that the results are displayed more clearly in the html report.